docs/src/architecture/passwordless_authentication.md

```diff
@@ -0,0 +1,41 @@
+<!--
+    Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+    SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Yubikey Passwordless Authentication
+
+This section describes Ghaf reference implementation for passwordless authentication for fast identity online (FIDO). The reference implementation has been created with Yubico Yubikeys.
+The implementation is modular and configurable - thus enabling also other implementations.
+
+This section describes Yubikey Passwordless Authentication and how to create U2F keys.
+
+The reader is expected to know the fundamentals of [FIDO](https://fidoalliance.org/specifications/), [Yubico](https://www.yubico.com/) and [U2F](https://developers.yubico.com/U2F/).
+
+## Prerequisites for Yubikey Passwordless Authentication
+
+User must have compatible [YubiKey hardware](https://www.yubico.com/products/) to begin with.
+
+## Creating Yubikey U2F Keys
+
+Yubikey U2F keys can be created with [pamu2fcfg](https://developers.yubico.com/pam-u2f/), a module implements PAM over U2F and FIDO2. pamu2fcfg is available in Nixpkgs as pkgs.pamu2fcfg.
+
+Use the following command to create your U2F keys for Ghaf gui-vm .
```

Comment:
So is the goal to eventually be able to set up the public key by running this within gui-vm?

Currently /etc/u2f_mappings is unwritable.

Reply:
Now the `/etc/u2f_mappings` should be link to `/nix/store/<hash>...` - meaning that the public key you set before build gets deployed to the target image and enabled. The previous implementation was so that user had to set up the public key by running the key generation process. We might be able to have both if Yubitools support multiple authorized keys files but let's not focus on that now. I requested @vunnyso to change it like it currently is that we can authorize public keys that have been generated before on another machine. This way we can have more secure authentication from the first guivm boot.